crates/goose/src/providers/formats/openai.rs

```diff
@@ -128,18 +326,53 @@ pub fn format_messages(messages: &[Message], image_format: &ImageFormat) -> Vec<
             }
         }
 
-        if converted.get("content").is_some() || converted.get("tool_calls").is_some() {
-            output.insert(0, converted);
+        if !converted["content"].is_null() || converted.get("tool_calls").is_some() {
+            messages_spec.push(converted);
         }
         messages_spec.extend(output);
     }
 
     messages_spec
 }
 
-/// Convert internal Tool format to OpenAI's API tool specification
-pub fn format_tools(tools: &[Tool]) -> anyhow::Result<Vec<Value>> {
-    let mut tool_names = std::collections::HashSet::new();
+/// Convert internal Tool format to OpenAI's API tool specification.
+///
+/// This function formats tools according to the OpenAI API tool specification.
+/// It handles tool names, descriptions, and parameters, ensuring they are properly
+/// formatted for the OpenAI API.
+///
+/// # Arguments
+/// * `tools` - The tools to format
+///
+/// # Returns
+/// * `anyhow::Result<Vec<Value>, Error>` - A vector of formatted tools ready for the OpenAI API,
+/// or an error if there are duplicate tool names
```

Comment:
```suggestion
///   or an error if there are duplicate tool names
```

^ this is why build is failing on`clippy` warning